### DISCUSSION TITLE
Heatmap over terrain layer

### DISCUSSION THREAD

**Discussion by Sgambe33** [visgl/deck.gl#9299] [Q&A]
Is it possible to draw maplibre heatmap or even deck.gl one over terrain layer? I tried but the heatmap remains below the terrain.

**Comment by Fabioni** [comment]
IMO most deck.gl layers don't work in combination with maplibre terrain. For icons you can manually set a height using `map. queryTerrainElevation` but for objects that are more than just a point, like bitmaplayers, linestrings, heatmaps etc. This does not work since you can't query every single point.

I hope there will a method to mitigate that in the future.

**Reply by Fabioni** [reply]
@Sgambe33 you could check the TerrainExtension, looks like what we want.
